### PR TITLE
jasmine: update class

### PR DESCRIPTION
**Pull request by UziTech**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://github.com/jasmine/jasmine-npm/blob/main/lib/jasmine.js>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by UziTech** [comment]
TODO:
- [ ] Rebase on #51997 after it is merged
- [ ] Add tests

**Review by RandinAk — APPROVED** [review]

**Comment by UziTech** [comment]
I'm not sure how to add tests for the jasmine class

**Comment by chivesrs** [comment]
So they don't have to be tests that work at runtime. Tests in this case means TypeScript either lets the code compile, or gives you a compile time error.

Meaning, you can do something like

```
jasmine.addSpecFiles(42); // $ExpectError (or whatever the keyword is, read the README for info)
jasmine.seed("string isn't a number"); // Error
jasmine.specDir; // $ExpectType string
```

Again, read the readme to find the exact syntax.

You don't need to be exhaustive for all the types, like don't create 1 test per new type you added, but give me enough so that the every day user of Jasmine can be assured the types are right.

Like, addMatchers is a popular function. Whereas (some other function) might not be as popular. So I'd want tests for addMatchers but not the other thing.

It's your judgment at the end of the day.

**Comment by UziTech** [comment]
The problem is that `jasmine` is the jasmine namespace not the class and there is no `addSpecFiles` etc. on the namespace.

**Comment by chivesrs** [comment]
I see. Let me clone and try.

File: types/jasmine/index.d.ts [hunk 1]
```diff
@@ -1032,36 +1032,45 @@ declare namespace jasmine {
...
+        requires: string[];
+        onCompleteCallbackAdded: boolean;
+        defaultReporterConfigured: boolean;
+
+        constructor(options: any);
         addMatchers(matchers: jasmine.CustomMatcherFactories): void;
         addReporter(reporter: jasmine.CustomReporter): void;
         addSpecFile(filePath: string): void;
         addSpecFiles(files: string[]): void;
+        addHelperFiles(files: string[]): void;
```
Comment (chivesrs) [inline types/jasmine/index.d.ts:1054]: Is this... still a real thing?
https://github.com/jasmine/jasmine/search?q=addHelperFiles

I feel like half of the things here no longer exist.

**Comment by chivesrs** [comment]
But, to answer your question, here's how I did it:

```
// tests based on http://jasmine.github.io/2.2/introduction.html

// NEW CODE BELOW
import JasmineClass from 'jasmine';

describe("Jasmine class", () => {
    const jasmine = new JasmineClass({});
    jasmine.stopOnSpecFailure(false);
});
// END NEW CODE

describe("A suite", () => {
    it("contains spec with an expectation", () => {
        expect(true).toBe(true);
    });
});
```

And then editing line 1570 following the directions about importing modules and how that declare statement changes, and then updating tsconfig to allow esModuleInterop: true.

I'm a bit undecided on if this jasmine "class" even exists any longer? And if it should be deleted? How do users instantiate said class? Can you provide a stackblitz that shows this class in action?

File: types/jasmine/index.d.ts (hunk 1 above)
Comment (UziTech) [reply types/jasmine/index.d.ts:1054]: The class is from [jasmine/jasmine-npm](https://github.com/jasmine/jasmine-npm)

https://github.com/jasmine/jasmine-npm/search?q=addHelperFiles

Comment (chivesrs) [reply types/jasmine/index.d.ts:1054]: Shouldn't that have its own DT package then?

Comment (chivesrs) [reply types/jasmine/index.d.ts:1054]: Never mind. I get it. It's an augmentation of browser jasmine to make it run on node, per https://jasmine.github.io/setup/nodejs.html.

I think you need to go with my solution below then.

**Comment by UziTech** [comment]
It looks like that doesn't work

**Comment by chivesrs** [comment]
You missed part of my directions.
Go back and reread what I said and find out what part you missed.

**Comment by UziTech** [comment]
It works! :tada:

**Review by chivesrs — APPROVED** [review]

**Comment by UziTech** [comment]
Ready to merge

**Pull request merged**